build.psm1

```diff
@@ -861,33 +861,35 @@ function New-PSOptions {
     Write-Verbose "Using framework '$Framework'"
 
     if (-not $Runtime) {
-        if ($environment.IsLinux) {
-            $Runtime = "linux-x64"
-        } elseif ($environment.IsMacOS) {
-            if ($PSVersionTable.OS.Contains('ARM64')) {
-                $Runtime = "osx-arm64"
-            }
-            else {
-                $Runtime = "osx-x64"
-            }
-        } else {
-            $RID = dotnet --info | ForEach-Object {
-                if ($_ -match "RID") {
-                    $_ -split "\s+" | Select-Object -Last 1
-                }
+        function Get-FirstMatch($String, $Pattern) {
+            $Selection = @($String | Select-String $Pattern)
+            if (-not $Selection) {
+                throw "Cannot match pattern: '${Pattern}'"
             }
+            $Selection[0].Matches.Groups[1].Value
+        }
 
+        try {
+            $Info = dotnet --info
+            Write-Verbose "dotnet --info:`n${Info}"
+
+            $Architecture = Get-FirstMatch $Info '^\s*Architecture:\s+(\w+)$'
+            $Platform = Get-FirstMatch $Info '^\s*OS Platform:\s+(\w+)$'
```

Comment:
```suggestion
            $Platform, $Architecture = $info |
                Select-String '^\s*OS Platform:\s+(\w+)$', '^\s*Architecture:\s+(\w+)$' |
                Select-Object -First 2 |
                ForEach-Object { $_.Matches.Groups[1].Value }
```

Reply:
Did not know you can pass multiple patterns to `Select-String`, nice.